### DISCUSSION TITLE
Does the .NET 8 IExceptionHandler support middleware pipeline re-execute?

### DISCUSSION THREAD

**Discussion by seangwright** [dotnet/aspnetcore#52096] [Q&A]
The previous ASP.NET Core exception handling approach supported [re-executing the middleware pipeline](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-7.0#exception-handler-page) based on the provided path.

```csharp
app.UseExceptionHandler("/Error");
```

This would set the request path to the new path and run the middleware pipeline again.

The benefit here is that an "error handled response" could be rendered via Controller, Razor Page, API endpoint.
Unfortunately this path was hardcoded, which mean you couldn't pick different paths at runtime based on the request context.

With `IExceptionHandler`, you can inspect the `HttpContext` but it isn't clear to me if its possible for the implementations to cause a pipeline re-execution. There aren't any docs on this feature yet, so I've been [reviewing the source code](https://source.dot.net/#Microsoft.AspNetCore.Diagnostics/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs,173).

Ideally, I'd like to be able to dynamically specify the path to re-execute based on the `Exception` using an `IExceptionHandler` implementation.

**Comment by dkj468** [comment]
@seangwright  With `IExceptionHandler` access to `HttpContext` could be helpful to redirect the request to some other endpoint. Example below : 

`httpContext.Response.Redirect("/data")`
